src/EditorFeatures/Core/Implementation/Workspaces/ProjectCacheService.SimpleMRUCache.cs

```diff
@@ -37,7 +37,7 @@ public bool Empty
             public void Touch(object instance)
             {
                 var oldIndex = -1;
-                var oldTime = DateTime.UtcNow;
+                var oldTime = DateTime.MaxValue;
```

Comment:
@heejaechang Can you give more context where the IndexOutOfRange was happening? Was `oldIndex = -1` at the end of the loop? If so, shouldn't we have seen an InvalidOperationException from the contract check before we hit IndexOutOfRange? If not, I don't see how `oldIndex` can be greater than `i`.